src/libraries/System.Private.CoreLib/src/System/Globalization/CharUnicodeInfo.cs

```diff
@@ -250,6 +250,77 @@ private static double GetNumericValueNoBoundsCheck(uint codePoint)
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static char ToUpper(char codePoint)
+        {
+            nuint offset = GetCategoryCasingTableOffsetNoBoundsChecks((uint)codePoint);
+
+            // The offset is specified in shorts:
+            // Get the 'ref short' corresponding to where the addend is, read it as a signed 16-bit value, then add
+
+            ref short rsStart = ref Unsafe.As<byte, short>(ref MemoryMarshal.GetReference(UppercaseValues));
+            ref short rsDelta = ref Unsafe.Add(ref rsStart, (nint)offset);
+            nint delta = (BitConverter.IsLittleEndian) ? rsDelta : BinaryPrimitives.ReverseEndianness(rsDelta);
```

Comment:
This should read `int delta`, not `nint delta`. The only time you need a cast to nint is in the call to `Unsafe.Add(..., (nint)offset)` above, since _offset_ is typed as `nuint`. (This means the code you have for line 262 is correct.)

Reply:
Will do that.